### PR TITLE
bugfix/DEVSU-2445 rapid report kb matches bug fix

### PR DESCRIPTION
**Pull request by bnguyen-bcgsc**
*Extension of [https://github.com/bcgsc/pori_ipr_client/pull/571]*

- DEVSU 2445 & 2467 & 2505
- Update kb matches and evidence level splitting logic in rapid report with new kb matched statements response data structure
- Update valueGetter logic for observed variants column in kb matches tables
- Switch positions of observed variants and known variants columns back to original
- Rework kbMatches statement categories handling in smallMutations, copyNumber, structuralVariants, and expressionOutliers to work with new kbmatches data structure
- Fix ExpOutliersType missing in GeneType

### CODE REVIEW

File: app/views/ReportView/components/RapidSummary/index.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const splitIprEvidenceLevels = (kbMatches: KbMatchType[]) => {
   return iprRelevanceDict;
 };
 
-const processPotentialClinicalAssociation = (variant: RapidVariantType) => Object.entries(getVariantRelevanceDict(variant.kbMatches))
+const processPotentialClinicalAssociation = (variant: any) => Object.entries(getVariantRelevanceDict(variant.kbMatches))
```
Comment (kttkjl) [inline app/views/ReportView/components/RapidSummary/index.tsx:60]: uhh ...?

Comment (bnguyen-bcgsc) [reply app/views/ReportView/components/RapidSummary/index.tsx:60]: srry typescript was throwing some errors I'll try to fix it

**Review by elewis2 — DISMISSED** [review]

File: app/views/ReportView/components/KbMatches/columnDefs.ts [hunk 2]
```diff
@@ -63,7 +63,21 @@ const columnDefs: ColDef[] = [
     sort: 'asc',
   },
   {
-    headerName: 'Observed Variant(s)',
+    headerName: 'Known Variants',
+    colId: 'kbVariant',
+    valueGetter: (params) => {
+      const { data: { kbMatches } } = params;
+      if (kbMatches) {
+        const kbVariants = kbMatches?.map((match) => match.kbVariant).filter((kbVariant) => kbVariant !== undefined);
```
Comment (kttkjl) [inline app/views/ReportView/components/KbMatches/columnDefs.ts:71]: we can probably just use `.filter` no?

`kbMatches?.filter((match) => match?.kbVariant);`

Comment (bnguyen-bcgsc) [reply app/views/ReportView/components/KbMatches/columnDefs.ts:71]: oh yea you're right let me simplify it

Comment (bnguyen-bcgsc) [reply app/views/ReportView/components/KbMatches/columnDefs.ts:71]: @kttkjl actually I tried it and it didn't work, we still have to extract kbVariant from each match with .map() right?

Comment (kttkjl) [reply app/views/ReportView/components/KbMatches/columnDefs.ts:71]: sorry, `match.kbVariant && match.kbVariant !== undefined`, maybe the shorthand is not enough

Comment (kttkjl) [reply app/views/ReportView/components/KbMatches/columnDefs.ts:71]: unless we want to keep `null, false, 0 and ''` values?

Comment (kttkjl) [reply app/views/ReportView/components/KbMatches/columnDefs.ts:71]: in which case we'll use reduce

```
kbMatches?.reduce((acc, match) => {
  if (match.kbVariant !== undefined) {
    acc.push(match.kbVariant);
  }
  return acc;
}, []);
```

**Review by kttkjl — APPROVED** [review]
Programmatically makes sense, just `reduce` vs `map` + `filter`

**Review by elewis2 — APPROVED** [review]

**Pull request merged**